File: backend/src/main/java/com/bookapp/model/ReadingProgress.java
package com.bookapp.model;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Document(collection = "reading_progress")
public class ReadingProgress {

    @Id
    private String id;

    private String userId;
    private String bookId;
    private String chapterId;

    private int chapterProgressPercent; 

    private LocalDateTime updatedAt;
}

File: BookApp/app/src/main/java/com/bookapp/data/model/ReadingProgress.kt
package com.bookapp.data.model

data class ReadingProgress(
    val id: String? = null,
    val userId: String? = null,
    val bookId: String? = null,
    val chapterId: String? = null,
    val chapterProgressPercent: Int? = 0,
    val updatedAt: String? = null
)
